types/connect/connect-tests.ts

```diff
@@ -1,45 +1,65 @@
+// npm i --no-save connect
+// tsc --noEmit false && node connect-tests.js
+
 import * as http from "http";
 import connect = require("connect");
 
 const app = connect();
 
 // log all requests
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    console.log(req, res);
+function logReq(req: connect.Request, res: http.ServerResponse, next: connect.NextFunction) {
+    console.log(`${req.method} ${req.originalUrl}`);
     next();
-});
-
-// "Throw" an Error
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    next(new Error("Something went wrong!"));
-});
-
-// "Throw" a number
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    next(404);
-});
+}
 
 // Stop on errors
-app.use((err: any, req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
+function stopOnErr(err: any, req: connect.Request, res: http.ServerResponse, next: connect.NextFunction) {
     if (err) {
-        return res.end(`Error: ${err}`);
+        res.statusCode = +err || 500;
+        res.end(`ERROR at ${req.url}: ${err}`);
+        return;
     }
-
     next();
+}
+
+app.use(logReq);
+
+// "Throw" a number
+app.use("/403", (req: connect.Request, res: http.ServerResponse, next: connect.NextFunction) => {
+    next(403);
 });
 
 // Use legacy `Function` for `next` parameter.
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: Function) => {
+// tslint:disable-next-line ban-types
+app.use((req: connect.Request, res: http.ServerResponse, next: Function) => {
     next();
 });
 
 // respond to all requests
-app.use((req: http.IncomingMessage, res: http.ServerResponse) => {
+app.use((req: connect.Request, res: http.ServerResponse) => {
     res.end("Hello from Connect!\n");
 });
 
-//create node.js http server and listen on port
+app.use(stopOnErr);
+
+// create node.js http server and listen on port
```

Comment:
Fine to clean up this comment, but I think that should be done separately to shrink the size of these changes.